osversion_android.go

```diff
@@ -1,87 +1,47 @@
 package osversion
 
-/*
-#include <sys/system_properties.h>
-#include <stdio.h>
-#include <stdlib.h>
-
-int android_get_prop_value_max() {
-   return PROP_VALUE_MAX;
-}
-
-int android_get_release(char *sdk_ver_str) {
-    int result = __system_property_get("ro.build.version.release", sdk_ver_str);
-    return result;
-}
-
-int android_get_api(char *sdk_ver_str) {
-    int result = __system_property_get("ro.build.version.sdk", sdk_ver_str);
-    return result;
-}
-*/
-import "C"
-
 import (
-	"errors"
 	"fmt"
+	"github.com/shirou/gopsutil/v3/host"
+	"os/exec"
 	"strconv"
-	"unsafe"
+	"strings"
 )
 
-func GetString() (string, error) {
-	strSize := int(C.android_get_prop_value_max())
-	verStr := (*C.char)(C.malloc(C.size_t(strSize)))
-	defer C.free(unsafe.Pointer(verStr))
-	C.android_get_release(verStr)
-	return C.GoString(verStr), nil
-}
-
-func GetHumanReadable() (string, error) {
-	// First get the semantic versioning scheme
-	verStr, err := GetString()
+func getAndroidAPI() (int, error) {
+	out, err := exec.Command("getprop", "ro.build.version.sdk").Output()
```

Comment:
Oh interesting, I didn't know that executing shell commands from an Android program was kosher. Are there any weird gotchas with doing this?

Reply:
Not on an emulator or on my phone (stock OS. Not rooted Google Pixel 5). Both worked just fine.

`getprop` is pretty common, I'd say, so all ROMS would need to support it